main/calculations.py

```diff
@@ -0,0 +1,147 @@
+    data = json.loads(response_json)
+    user = request.user
+    if user.email == data['email']:
+        error_message = 'It is your existing email address'
+        messages.error(request, error_message)
+    else:
+        user.email = data['email']
+        user.save()
+        success_message = f'You changed your email. Every messages will be send to \
+                    {request.user.email} until now.'
+        messages.success(request, success_message)
+
+
+class SiteDownChecker:
+
+    def __init__(self, url, user):
+        self.url = url
+        self.time = 0
+        self.error = None
+        self.user = user
+
+    def status(self, proxy=False):
+        try:
+            if proxy:
+                r = ProxyRequests(self.url)
+            else:
+                r = requests.get(self.url, headers={
+                    'User-Agent': 'Mozilla/4.0 (compatible; MSIE 9.0; Windows NT 6.1)'})
+            if SiteToCheck.objects.filter(url=self.url, user=self.user).exists():
+                return self.modify_url_success(proxy, r)
+            else:
+                return self.create_new_url_success(proxy, r)
+        except Exception as e:
+            self.error = str(e)
+            if not proxy and config.PROXY:
+                return self.status(proxy=True)
+            if SiteToCheck.objects.filter(url=self.url, user=self.user).exists():
+                return self.modify_url_exception(e)
+            else:
+                return self.create_url_exception()
+
+    def create_url_exception(self):
+        data = dict()
+        SiteToCheck.objects.create(url=self.url,
+                                   user=self.user,
+                                   last_status=None,
+                                   last_response_time=None,
+                                   last_check=datetime.now().strftime("%Y-%m-%d %H:%M"),
+                                   bad_data=str(
+                                       datetime.now().strftime("%Y-%m-%d %H:%M")) + ': The url is not responding'
+                                   )
+        data['last_status'] = None
+        data['last_response_time'] = None
+        data['bad_data'] = 'The url is not responding'
+        return data
+
+    def modify_url_exception(self, e):
+        data = dict()
+        obj = SiteToCheck.objects.get(url=self.url, user=self.user)
+        obj.last_status = None
+        obj.last_response_time = None
+        obj.last_check = datetime.now().strftime("%Y-%m-%d %H:%M")
+        if len(obj.bad_data) > 0:
+            obj.bad_data += '\n'
+        obj.bad_data += str(datetime.now().strftime("%Y-%m-%d %H:%M")) + ': ' + self.error
+        obj.save()
+        data['bad_data'] = e
+        data['last_status'] = None
+        data['last_response_time'] = None
+        data['last_check'] = datetime.now().strftime("%Y-%m-%d %H:%M")
+        data['url'] = self.url
+        return data
+
+    def create_new_url_success(self, proxy, r):
+        data = dict()
```

Comment:
Raczej nie tworzy się tak słowników z gotowymi danymi.
```
a = dict()
a['a'] = 'b'
```
Częściej robi się to tak:
```
a = {'a': 'b'}
```